composer.json

```diff
@@ -46,51 +31,58 @@
         "phpcr/phpcr-shell": "~1.0",
         "symfony/debug": "^3.4",
         "symfony/debug-bundle": "^3.4",
+        "symfony/dotenv": "^3.4",
         "symfony/inflector": "^3.4",
         "symfony/phpunit-bridge": "^3.4",
         "symfony/var-dumper": "^3.4",
         "symfony/web-profiler-bundle": "^3.4",
         "symfony/web-server-bundle": "^3.4"
     },
+    "autoload": {
+        "psr-4": {
+            "App\\": "src/"
+        }
+    },
+    "autoload-dev": {
+        "psr-4": {
+            "App\\Tests\\": "tests/"
+        }
+    },
+    "replace": {
+        "symfony/polyfill-ctype": "*",
+        "symfony/polyfill-iconv": "*",
+        "symfony/polyfill-php71": "*",
+        "symfony/polyfill-php70": "*",
+        "symfony/polyfill-php56": "*"
+    },
     "scripts": {
-        "post-install-cmd": [
-            "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
-            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
-            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
-            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
-            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile",
-            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::prepareDeploymentTarget",
+        "auto-scripts": [
+            "bin/adminconsole cache:clear",
+            "bin/websiteconsole cache:clear",
+            "bin/adminconsole assets:install --symlink --relative",
             "Sulu\\Bundle\\MediaBundle\\Composer\\MediaScriptHandler::initBundle",
             "Massive\\Bundle\\SearchBundle\\Composer\\SearchScriptHandler::initBundle"
         ],
+        "post-install-cmd": [
+            "@auto-scripts"
+        ],
         "post-update-cmd": [
-            "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
-            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
-            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
-            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
-            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile",
-            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::prepareDeploymentTarget",
-            "Sulu\\Bundle\\MediaBundle\\Composer\\MediaScriptHandler::initBundle",
-            "Massive\\Bundle\\SearchBundle\\Composer\\SearchScriptHandler::initBundle"
+            "@auto-scripts"
         ],
         "post-create-project-cmd": [
             "Sulu\\Bundle\\CoreBundle\\Composer\\ScriptHandler::removeComposerLockFromGitIgnore",
             "Sulu\\Bundle\\CoreBundle\\Composer\\ScriptHandler::removePackageLockJsonFromGitIgnore"
         ]
     },
     "config": {
-        "bin-dir": "bin"
+        "bin-dir": "bin",
+        "preferred-install": {
+            "*": "dist"
```

Comment:
@danrot I configured this in my global composer configuration:

`vim ~/.composer/config`

```
{
    "config": {
        "preferred-install": {
            "sulu/*": "source",
            "massive/*": "source"
        }
    }
}
```

and did work for me also in this case.

Reply:
Hm... if we want people to contribute that seems like a major obstacle. What's so bad about keeping it the way we've had? Or at least make our namespaces "auto"?

Reply:
@danrot I don't think that auto does change anything about contributing as it will still do dist by default then. But its ok for me when we remove this configuration. I only added it to be similiar to symfony composer.json.